coordinator/src/main/java/com/tencent/rss/coordinator/PartitionBalanceAssignmentStrategy.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Tencent is pleased to support the open source community by making
+ * Firestorm-Spark remote shuffle server available.
+
+ * Copyright (C) 2021 THL A29 Limited, a Tencent company.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * https://opensource.org/licenses/Apache-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.tencent.rss.coordinator;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.tencent.rss.common.PartitionRange;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+public class PartitionBalanceAssignmentStrategy implements AssignmentStrategy {
+
+  private ClusterManager clusterManager;
+  private Map<ServerNode, PartitionInfo> serverToPartitions = Maps.newConcurrentMap();
+
+  public PartitionBalanceAssignmentStrategy(ClusterManager clusterManager) {
+    this.clusterManager = clusterManager;
+  }
+
+  @Override
+  public PartitionRangeAssignment assign(
+      int totalPartitionNum,
+      int partitionNumPerRange,
+      int replica,
+      Set<String> requiredTags) {
+
+    if (partitionNumPerRange != 1) {
+      throw new RuntimeException("PartitionNumPerRange must be one");
+    }
+
+    SortedMap<PartitionRange, List<ServerNode>> assignments = new TreeMap<>();
+    synchronized (this) {
+        List<ServerNode> nodes = clusterManager.getServerList(requiredTags);
+        Map<ServerNode, PartitionInfo> newPartitionInfos = Maps.newConcurrentMap();
+        for (ServerNode node : nodes) {
+          PartitionInfo partitionInfo;
+          if (serverToPartitions.containsKey(node)) {
+            partitionInfo = serverToPartitions.get(node);
+            if (partitionInfo.getTimestamp() < node.getTimestamp()) {
+              partitionInfo.resetPartitionNum();
+              partitionInfo.setTimestamp(node.getTimestamp());
+            }
+          } else {
+            partitionInfo = new PartitionInfo();
+          }
+          newPartitionInfos.putIfAbsent(node, partitionInfo);
+        }
+        serverToPartitions = newPartitionInfos;
+        int assignPartitions = totalPartitionNum * replica / clusterManager.getShuffleNodesMax();
```

Comment:
assignPartitionNum should be great equal than 1

Reply:
Fix